src/js/sequences.js

```diff
@@ -0,0 +1,306 @@
+      break;
+    }
+  }
+
+  // the first two periods are trivial
+  dd.shift();
+  dd.shift();
+
+  jQuery("#info_rank_2_mos").text(dd.join(", "));
+}
+
+// helper function to simply pass in an interval and get an array of ratios returned
+function get_rational_approximations(intervalIn, numerators, denominators, roundf=999999,
+cidxOut=null, ratiosOut=null, numlimits=null, denlimits=null, ratiolimits=null) {
+  var cf = []; // continued fraction
+
+  cf = get_cf(intervalIn, 15, roundf);
+  get_convergents(cf, numerators, denominators, roundf, cidxOut);
+
+  var doRatios = !(ratiosOut===null);
+  var doNumLim = !(numlimits===null);
+  var doDenLim = !(denlimits===null);
+  var doRatioLim = !(ratiolimits===null);
+
+  if (doRatios|| doNumLim || doDenLim || doRatioLim) {
+    var nlim;
+    var dlim;
+
+    for (let i = 0; i < numerators.length; i++) {
+      numerators[i] === 1 ? nlim = 1 : nlim = get_prime_limit(numerators[i]);
+      denominators[i] === 1 ? dlim = 1 : dlim = get_prime_limit(denominators[i]);
+
+      if (doRatios)
+        ratiosOut.push(numerators[i]+"/"+denominators[i]);
+      if (doNumLim)
+        numlimits.push(nlim);
+      if (doDenLim)
+        denlimits.push(dlim);
+      if (doRatioLim)
+        ratiolimits.push(Math.max(nlim, dlim));
+    }
+  }
+}
+
+// rank2 scale algorithm intended for integers, in ET contexts
+// for example, period = 12, gen = 7 : [ 2 2 1 2 2 2 1 ]
+function get_rank2_mode(period, generator, size, numdown=0) {
+  let degrees = [];
+  let modeOut = [];
+  var interval;
+
+  interval = generator * -numdown;
+  for (let n = 0; n < size; n++) {
+    while (interval < 0) {
+      interval += period;
+    }
+    if (interval >= period) {
+      interval %= period;
+    }
+    degrees.push(interval);
+    interval += generator;
+  }
+
+  degrees.sort(function(a, b) { return a-b });
+  for (let n = 1; n < degrees.length; n++) {
+    modeOut.push(degrees[n] - degrees[n-1]);
+  }
+
+  modeOut.push(period - degrees[degrees.length-1]);
+
+  return modeOut;
+}
+
+// returns an array representing the prime factorization
+// indicies are the 'nth' prime, the value is the powers of each prime
+function get_prime_factors(number) {
+  number = Math.floor(number);
+  if (number === 1) {
+    //alert("Warning: 1 has no prime factorization.");
+    return 1;
+   }
+  var factorsout = [];
+  var n = number;
+  var q = number;
+  var loop;
+
+  for (let i = 0; i < PRIMES.length; i++) {
+    if (PRIMES[i] > n)
+      break;
+
+    factorsout.push(0);
+
+    if (PRIMES[i] === n) {
+      factorsout[i]++;
+      break;
+    }
+
+    loop = true;
+
+    while (loop) {
+      q = n / PRIMES[i];
+
+      if (q === Math.floor(q)) {
+        n = q;
+        factorsout[i]++;
+        continue;
+      }
+      loop = false;
+    }
+  }
+
+  return factorsout;
+}
+
+ // returns an array of integers that share no common factors to the given integer
+ function get_coprimes(number) {
+  let coprimes = [1];
+  var m, d, t;
```

Comment:
These variables might need rethinking in the future, because these don't give any information on what these are for or what values are stored in them.